geode-core/src/integrationTest/java/org/apache/geode/internal/cache/DiskRegionCompactorClearsObjectThatAreNoLongerNeededIntegrationTest.java

```diff
@@ -0,0 +1,455 @@
+   * 2. Execute destroy operation for every fifth entry, and each time add new entry. This will
+   * result with few additional Oplog files. Compaction threshold will not be reached.
+   * 3. Destroy all operations created in step 2. This will trigger compaction of files that
+   * were created in step 2. Compaction will delete only .crf and .krf files, but will not
+   * delete .drf files because they contain destroy operations for events located in
+   * .crf files crated in step 1. Check that unnecessary objects are cleared for the
+   * Oplog that represents orphaned .drf file (no accompanying .crf and .krf file)
+   **/
+  @Test
+  public void testCompactorRegionMapDeletedForOnlyDrfOplogAfterCompactionIsPerformed()
+      throws InterruptedException {
+
+    final int ENTRY_RANGE_1 = 300;
+    final int ENTRY_RANGE_2 = 600;
+
+    createDiskStore(30, 10000);
+    Region<Object, Object> region = createRegion();
+    DiskStoreImpl diskStore = ((LocalRegion) region).getDiskStore();
+
+    // Create several oplog files (.crf and .drf) by executing put operations in defined range
+    for (int i = 0; i < ENTRY_RANGE_1; i++) {
+      region.put(i, new byte[100]);
+    }
+    await().untilAsserted(() -> assertThat(getCurrentNumberOfOplogs(diskStore)).isEqualTo(5));
+
+    // Destroy every fifth entry from previous range and each time put new entry in new range.
+    // This will create additional oplog files (.crf and .drf), but compaction will not be triggered
+    // as threshold will not be reached. Oplog files (.drf) created in this step will contain
+    // destroys for events that are located in .crf files from previous range.
+    TombstoneService tombstoneService = ((InternalCache) cache).getTombstoneService();
+    int key = 0;
+    while (key < ENTRY_RANGE_1) {
+      region.destroy(key);
+      // It is necessary to force tombstone expiration, otherwise event won't be stored in .drf file
+      // and total live count won't be decreased
+      await().untilAsserted(
+          () -> assertThat(tombstoneService.forceBatchExpirationForTests(1)).isTrue());
+      region.put(key + ENTRY_RANGE_1, new byte[300]);
+      key = key + 5;
+    }
+    await().untilAsserted(() -> assertThat(getCurrentNumberOfOplogs(diskStore)).isEqualTo(7));
+
+    // Destroy all events created in previous step in order to trigger automatic compaction.
+    // This will trigger compaction for the files that were created in previous step.
+    // Compaction will delete .crf and .krf file, but will leave .drf file because it contains
+    // destroy operation for the events that are located in some older .crf files.
+    key = ENTRY_RANGE_1;
+    while (key < ENTRY_RANGE_2) {
+      region.destroy(key);
+      assertThat(tombstoneService.forceBatchExpirationForTests(1)).isTrue();
+      key = key + 5;
+    }
+
+    // wait for all Oplog's to be compacted
+    await().untilAsserted(() -> assertThat(isOplogToBeCompactedAvailable(diskStore)).isFalse());
```

Comment:
Seems to me that this part of the test case is repeated in three test cases. Could this code be taken out to a method so that the code is reused across the test cases?

Reply:
All comments applied. I have put each step in separate function and use those functions in each tests. I think it will be more readable this way. Thanks for the review.